src/applications/third-party-apps/containers/SearchForm/index.jsx

```diff
@@ -0,0 +1,148 @@
+// Node modules.
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import URLSearchParams from 'url-search-params';
+import map from 'lodash/map';
+import { connect } from 'react-redux';
+// Relative imports.
+import CATEGORIES from '../../constants/CATEGORIES';
+import PLATFORMS from '../../constants/PLATFORMS';
+import scrollToTop from 'platform/utilities/ui/scrollToTop';
+import { fetchResultsThunk } from '../../actions';
+
+export class SearchForm extends Component {
+  static propTypes = {
+    // From mapStateToProps.
+    fetching: PropTypes.bool.isRequired,
+    // From mapDispatchToProps.
+    fetchResultsThunk: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    // Derive the current name params.
+    const queryParams = new URLSearchParams(window.location.search);
+
+    // Derive the state values from our query params.
+    const category = queryParams.get('category') || '';
+    const platform = queryParams.get('platform') || '';
```

Comment:
Seems like there's a lot of reference to the query params and some of the keys/values feel a little fragile when things start to change. What do you think of creating a layer of abstraction by centralizing access to the query params in its own module?